doc/dap.txt

```diff
@@ -644,6 +644,91 @@ toggle_multiline_display({value})  *dap.ui.variables.toggle_multiline_display()*
         in a hover window.
 
 
+
+==============================================================================
+WIDGET API                                                    *dap-widgets*
+
+The UI of nvim-dap is by default minimal and noninvasive, but it provides
+widget primitives that can be used to build and customize a UI.
+
+Some examples:
+
+
+View the current scopes in a sidebar:
+
+>
+  local widgets = require('dap.ui.widgets')
+  local my_sidebar = widgets.sidebar(widgets.scopes)
+  my_sidebar.open()
+<
+
+View the current frames in a sidebar:
+
+>
+  local widgets = require('dap.ui.widgets')
+  local my_sidebar = widgets.sidebar(widgets.frames)
+  my_sidebar.open()
+<
+
+
+View the frames in a centered floating window:
+
+>
+  local widgets = require('dap.ui.widgets')
+  widgets.centered_float(widgets.scopes)
+<
+
+
+View the value for the expression under the cursor in a floating window:
+
+>
+  require('dap.ui.widgets').hover()
+<
+
+
+Available widgets entities:
+
+- scopes
+- frames
+- expression
+
+
+Available widget builder functions:
+
+- sidebar({widget})
+    Creates a view for a sidebar. You must call `open` on the result to open the window.
+
+- cursor_float({widget})
+    Opens the contents of the widget in a floating window anchored at the cursor.
+
+- centered_float({widget}
```

Comment:
Would probably be a low hanging fruit to add a `winopts` argument here - similar to how `repl.open` works?